### DISCUSSION TITLE
Confirm the license when using parts modification  of  bit7z

### DISCUSSION THREAD

**Discussion by msclock** [rikyoz/bit7z#212] [Q&A] [answered]
Hi, I found bit7z has simplier and more elegant wrapperd interfaces from 7zip. Great work.

I maybe use some modification parts of bit7z for https://github.com/msclock/smooth7zip. And do I have to use the [Mozilla Public License v2](https://www.mozilla.org/en-US/MPL/2.0/) from https://github.com/rikyoz/bit7z/issues/86 instead of the more permissive Apache Software License or the MIT license? Or I can  keep MPL-2 in files and use Apache license.

Thanks.😀

**Comment by rikyoz** [comment, marked as answer]
Hi!

> I found bit7z has simplier and more elegant wrapperd interfaces from 7zip. Great work.

Thank you!

> I maybe use some modification parts of bit7z for https://github.com/msclock/smooth7zip. And do I have to use the [Mozilla Public License v2](https://www.mozilla.org/en-US/MPL/2.0/) from https://github.com/rikyoz/bit7z/issues/86 instead of the more permissive Apache Software License or the MIT license? Or I can keep MPL-2 in files and use Apache license.

I am not a lawyer, so take this with some scepticism: as I understand it, any modification of MPLv2 software such as bit7z must be made available under the MPLv2, i.e., it _cannot_ be re-licensed under a different license.

However, _unmodified_ MPL-2.0 code may be freely combined with code under another license to form a larger work.

Anyway, if you think your modifications might be useful to the users of bit7z, feel free to open a PR. I'll be happy to review it to see if it makes sense to include your changes in the library.

**Comment by msclock** [comment]
I see, and thanks for clarification. What I want to do is make it header-only simply and use the compiled and patched 7zip lib I have exported some necessary interfaces. I mean this is a big breaking change and I don't confirm it is suitable as a pr to bit7z.

**Reply by rikyoz** [reply]
>  I mean this is a big breaking change and I don't confirm it is suitable as a pr to bit7z.

Yeah, in this case I don't think it's suitable for a PR, I'm sorry!
